src/Calculator.php

```diff
@@ -23,12 +23,24 @@ public function __construct(ComposerFile $composer, PackagistAPI $packagist)
     public function getDependencyInfo(string $directory): array
     {
         $dependencies = $this->composer->getDependencies($directory);
+        $repositories = array_map(
+			fn($repositoryUrl) => $this->packageAPI->getRepositoryInfo($repositoryUrl),
+			$this->composer->getRepositoriesUrl($directory) ?: ['https://repo.packagist.org']
+		);
 
         foreach ($dependencies as $dependency) {
-            $package_info = $this->packagist->getPackageInfo($dependency->name);
-            if (empty($package_info)) {
-                continue;
-            }
+			foreach ($repositories as $repository) {
+				if ($repository == null || !$repository->hasPackage($dependency->name)) {
+					continue;
+				}
+				$package_info = $this->packageAPI->getPackageInfo($dependency->name, $repository->getPackageUrl($dependency->name));
+				if (!empty($package_info)) {
+					break;
+				}
+			}
```

Comment:
Based on my understanding of your goal, an iterator should be better.

```suggestion
                        $repositoriesIterator = new ArrayIterator($repositories);
			while(empty($package_info) && $repositoriesIterator->valid()) {
				if ($repositoriesIterator->current()->hasPackage($dependency->name)) {
					$package_info = $this->packageAPI
					    ->getPackageInfo(
					       $dependency->name,
					       $repositoriesIterator->current()->getPackageUrl($dependency->name)
					);
				}
				
				$repositoryIterator->next()
			}
```

Keep in mind that you could instantiate `$repositoriesIterator` before the loop on `$dependencies` and `$repositoriesIterator->rewind()` to reset it at the beginning of the loop.

The same logic could be applied with `$dependencies`

Reply:
Done :) Thank you for the help